crates/symbolicator-service/src/caching/memory.rs

```diff
@@ -53,13 +54,23 @@ impl<T: CacheItemRequest> Clone for Cacher<T> {
 
 impl<T: CacheItemRequest> Cacher<T> {
     pub fn new(config: Cache, shared_cache: SharedCacheRef) -> Self {
-        // TODO: eventually hook up configuration to this
-        // The capacity(0) and ttl(0) make sure we are not (yet) reusing in-memory caches, except
-        // for request coalescing right now.
         let cache = InMemoryCache::builder()
-            .max_capacity(0)
-            .time_to_live(Duration::ZERO)
+            .max_capacity(config.in_memory_capacity)
+            .name(config.name().as_ref())
+            // NOTE: even though we have a per-item TTL, we still want to have a hard limit here
+            // FIXME: we have a rather conservative hard limit right now before we have properly
+            // immutable caches, and to slowly ramp this up
+            .time_to_live(Duration::from_secs(5 * 60))
+            // NOTE: we count all the bookkeeping structures to the weight as well
+            .weigher(|_k, v| {
+                let value_size =
+                    v.1.as_ref()
+                        .map_or(0, |item| T::weight(item))
```

Comment:
I think this can be written point-free
```suggestion
                        .map_or(0, T::weight)
```